gov2/redshift/scenarios/redshift_basics.go

```diff
@@ -0,0 +1,378 @@
+		deleted, err = runner.redshiftActor.DeleteCluster(ctx, clusterId)
+		if err != nil {
+			log.Printf("Error deleting cluster: %v", err)
+		}
+	}
+	if !deleted && questioner.AskBool("Do you want to delete the dev table? This will clean up all inserted records but keep your cluster intact. (y/n)", "y") {
+		deleted, err = runner.redshiftDataActor.DeleteTable(ctx, clusterId, databaseName, tableName, userName)
+		if err != nil {
+			log.Printf("Error deleting movies table: %v", err)
+		}
+	}
+	if !deleted && questioner.AskBool("Do you want to delete all rows in the Movies table? This will clean up all inserted records but keep your cluster and table intact. (y/n)", "y") {
+		deleted, err = runner.redshiftDataActor.DeleteDataRows(ctx, clusterId, databaseName, tableName, userName, runner.pauser)
+		if err != nil {
+			log.Printf("Error deleting data rows: %v", err)
+		}
+	}
+	if !deleted {
+		log.Print("Please manually delete any unwanted resources.")
+	}
+}
+
+// snippet-end:[gov2.redshift.BasicsScenario]
+
+// snippet-start:[gov2.redshift.loadMoviesFromJSON]
+
+// loadMoviesFromJSON takes the <fileName> file and populates a slice of Movie objects.
+func (runner *RedshiftBasicsScenario) loadMoviesFromJSON(fileName string, filesystem demotools.IFileSystem) ([]Movie, error) {
+	file, err := filesystem.OpenFile("../../resources/sample_files/" + fileName)
+	if err != nil {
+		return nil, err
+	}
+	defer filesystem.CloseFile(file)
+
+	var movies []Movie
+	err = json.NewDecoder(file).Decode(&movies)
+	if err != nil {
+		return nil, err
+	}
+
+	return movies, nil
+}
+
+// snippet-end:[gov2.redshift.loadMoviesFromJSON]
+
+// snippet-start:[gov2.redshift.WaitForClusterAvailable]
+
+// WaitForClusterAvailable loops until a success or failure state is returned about the given cluster.
+func (runner *RedshiftBasicsScenario) WaitForClusterAvailable(client *redshift.Client, clusterId string, questioner demotools.IQuestioner, pauser demotools.IPausable) error {
+	questioner.Ask("Now we will wait until the cluster is available. Press Enter to continue...")
+
+	log.Println("Waiting for cluster to become available. This may take a few minutes.")
+	describeClusterInput := &redshift.DescribeClustersInput{
+		ClusterIdentifier: &clusterId,
+	}
+
+	for {
+		// snippet-start:[gov2.redshift.DescribeClusters]
+		output, err := client.DescribeClusters(context.TODO(), describeClusterInput)
+		if err != nil {
+			return err
+		}
+
+		if output.Clusters[0].ClusterStatus != nil && *output.Clusters[0].ClusterStatus == "available" {
+			log.Println("Cluster is available! Total Elapsed Time:", time.Since(time.Now().Add(-1*time.Second*time.Duration(output.Clusters[0].ClusterCreateTime.Second()))))
+			return nil
+		}
+
+		log.Print("Elapsed Time: ")
+		log.Println(time.Since(time.Now().Add(-1 * time.Second * time.Duration(output.Clusters[0].ClusterCreateTime.Second()))))
+		// snippet-end:[gov2.redshift.DescribeClusters]
```

Comment:
Don't snippet interior code like this.
DescribeClusters should be extracted into an action and given the full treatment to make it a real snippet.

Reply:
Extracted